packages/controllers/jest.config.js

```diff
@@ -8,10 +8,10 @@ module.exports = {
   coveragePathIgnorePatterns: ['/node_modules/', '/mocks/', '/test/'],
   coverageThreshold: {
     global: {
-      branches: 54,
-      functions: 70,
-      lines: 66,
-      statements: 66,
+      branches: 57,
```

Comment:
How do we determine what is an appropriate threshold and why these specific numbers?

Reply:
For present purposes, the simple answer is "more than what they are at the moment", and the tests I added here and in other PRs raised them beyond the previous limit we had set.

Aiming for 100 for all of them is generally a good idea, although it's important to not obsess too much over coverage metrics. @Gudahtt recently shared an article with me discussing the relationship between coverage and test suite quality, and it concluded that coverage is not that important once you account for the _size_ of the test suite, although I would guess that the size of the test suite and its coverage metrics are highly correlated.

Reply:
The sub-100 coverage thresholds we use throughout all of our repos are attempts to ratchet up the minimum coverage to ensure it doesn't get worse. 100 is always the goal though.

Here's the article Erik mentions: https://neverworkintheory.org/2021/09/24/coverage-is-not-strongly-correlated-with-test-suite-effectiveness.html
It basically showed that "number of tests" was a better measure of the effectiveness of a test suite than test coverage was. My takeaway is that 100% coverage is helpful but insufficient. Better to test all cases you can think of, beyond just those you need for coverage. And also to adopt policies like including a test with every bug fix.